mava/components/jax/building/data_server.py

```diff
@@ -99,12 +99,155 @@ def name() -> str:
         return "data_server"
 
 
+@dataclass
+class SamplerConfig:
+    pass
+
+
+class Sampler(Component):
+    def __init__(
+        self,
+        config: SamplerConfig = SamplerConfig(),
+    ):
+        """_summary_
+
+        Args:
+            config : _description_.
+        """
+        self.config = config
+
+    @abc.abstractmethod
+    def on_building_data_server_start(self, builder: SystemBuilder) -> None:
+        """_summary_"""
+
+    @staticmethod
+    def name() -> str:
+        """_summary_
+
+        Returns:
+            _description_
+        """
+        return "data_server_sampler"
+
+    @staticmethod
+    def config_class() -> Optional[Callable]:
+        """Config class used for component.
+
+        Returns:
+            config class/dataclass for component.
+        """
+        return SamplerConfig
+
+
+class UniformSampler(Sampler):
+    def on_building_data_server_start(self, builder: SystemBuilder) -> None:
+        """_summary_"""
```

Comment:
Everything needs proper docstrings please.

Reply:
The rate limiters have no docstrings so I'm not exactly sure the format we are expected to follow.

Reply:
This is the format used elsewhere:

"""Short description of what method does.

    Further explanation of method, if needed.

    Args:
        arg1: Explanation about what it is.

    Returns:
        Return type here.
"""

Reply:
No other components have docs and i probably will write them incorrectly and they will have to change anyway. I honestly think docs should be written by the mava team.